### ISSUE TITLE
Decouple schema fetch queries from server-side timeouts

### ISSUE THREAD

**Issue by avikivity** [scylladb/java-driver#293]
If the user set server-side timeouts, or service level timeouts, this impacts schema fetches. The timeout can be too low to allow a schema fetch when the schema is very large.

Instead, qualify schema (and other metadata) fetches with `USING TIMEOUT` (with a high and configurable default) so schema fetches work regardless of timeout tuning.

**Comment by avikivity** [comment]
Also see https://github.com/scylladb/scylladb/issues/18471

**Comment by roydahan** [comment]
Don't we need it in all drivers? (this one is specific to java driver)

**Comment by mykaul** [comment]
> Don't we need it in all drivers? (this one is specific to java driver)

Most likely.

**Comment by mykaul** [comment]
> Implementation note: `USING TIMEOUT` is not supported in older version of Scylla and in all versions of Cassandra, so probably the code would have to take that into account (but it's a bit annoying to find out if the cluster supports this).

What's our compatibility guarantee for old versions?

**Comment by roydahan** [comment]
Maybe we should have this issue at scylladb core and clone it to all drivers?

**Comment by mykaul** [comment]
> Maybe we should have this issue at scylladb core and clone it to all drivers?

I don't mind. I'd be happy to see implementation in one, then follow-up on the rest.

**Comment by roydahan** [comment]
> I don't mind. I'd be happy to see implementation in one, then follow-up on the rest.

I agree, but I would like to start having a better process to track such issues/features.
I thought of having the RFE in one repo (e.g. scylladb OSS) and links to all relevant driver issues from there.
